src/main/java/uk/gov/companieshouse/api/accounts/controller/AccountController.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.companieshouse.api.accounts.controller;
+
+import javax.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+import uk.gov.companieshouse.api.accounts.model.rest.Account;
+import uk.gov.companieshouse.api.accounts.service.AccountService;
+
+@RestController
+public class AccountController {
+
+    @Autowired
+    private AccountService accountService;
+
+    @PostMapping(value = "/transactions/{transactionId}/accounts",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity createAccount(@Valid @RequestBody Account account) {
+        account = accountService.createAccount(account);
+
+        if (account != null) {
```

Comment:
Should we be checking for null here? By explicitly checking for null, aren't we coupling our implementation of a lower level (I think here it's the Service implementation, in which case you're always allocating the object in memory so it would never be null?)

Reply:
removed. forgot that was there, it was originally in the service layer for the mongo response, but it only ever responds with the inserted object or a runtime exception.